spectator/sidecarmeter.py

```diff
@@ -0,0 +1,30 @@
+import re
+
+from spectator.id import MeterId
+
+
+class SidecarMeter:
+    ALLOWED_CHARS = re.compile("[^-._A-Za-z0-9~^]")
+
+    def __init__(self, meter_id: MeterId, meter_type: str):
+        """
+        :param meter_id:
+            Base identifier for all measurements supplied by this meter.
+        :param meter_type:
+            Prefix string for line to output to SpectatorD.
+        """
+        self.meterId = meter_id
+        self.idString = self._create_id_string(meter_id, meter_type)
+
+    def _replace_invalid_chars(self, s: str) -> str:
+        return self.ALLOWED_CHARS.sub("_", s)
+
+    def _create_id_string(self, meter_id: MeterId, meter_type: str) -> str:
+        s = "{}:{}".format(meter_type, self._replace_invalid_chars(meter_id.name))
+
+        for key, value in sorted(meter_id.tags().items()):
```

Comment:
There was a `sanitize` method added to the id class, any reason it wasn't used here? It seems like the sanitization logic is a bit different between the two and in two places.

Reply:
This was leftover from the previous iteration where I was using the `__str__` method on `id`s to generate the protocol line from SpectatorD. I think this logic is better contained in the `SidecarMeter` class, which is now the base class for all meter classes. This split also opens the possibility of inspecting the `MeterId` to discover which bad characters have been rewritten to underscores, in case there are questions on it. I reverted the `id` string logic to the previous version.